crates/matrix-sdk-ui/src/timeline/event_item/content.rs

```diff
@@ -403,6 +403,20 @@ impl ReactionGroup {
     pub fn senders(&self) -> impl Iterator<Item = &UserId> {
         self.values().unique().map(AsRef::as_ref)
     }
+
+    /// All reactions within this reaction group that were sent by the given
+    /// user.
+    ///
+    /// Note that it is possible for multiple reactions by the same user to
+    /// have arrived over federation.
+    pub fn by_sender<'a>(
+        &'a self,
+        user_id: &'a UserId,
+    ) -> impl Iterator<Item = (Option<&OwnedTransactionId>, Option<&OwnedEventId>)> + 'a {
+        self.iter()
+            .map(|(k, v)| ((k.0.as_ref(), k.1.as_ref()), v))
+            .filter_map(move |(k, v)| (*v == user_id).then_some(k))
```

Comment:
I think it should be possible to combine these into one `filter_map`?

Reply:
[`4fe2968`](https://github.com/matrix-org/matrix-rust-sdk/pull/2118/commits/4fe2968306a6ba21bfd18be13e4dcff9655a0790)